### ISSUE TITLE
Purchasing/Selling Items

### ISSUE THREAD

**Issue by MarvelProgramming** [MarvelProgramming/BetterTrader#11]
Items should be purchasable/sellable from the trader, based on BT's configuration. Doing so requires the player have the necessary coins to make the purchase and that the player have an available inventory slot for the newly purchased item (or a slot with the same item and enough space left to stack what the player purchased). Upon purchasing an item, it should be placed in the player's inventory, and coins removed from the player based on the total value of the purchase. After the purchase, the UI should refresh to reflect the updated state of the trader and player (i.e. their coins and inventories should be mirrored correctly).

#### In summary:
- Update UI after purchase/sale
- Require player have necessary coin and inventory space
- Items should appear based on BT configuration
